cetmix_tower_server/models/cx_tower_server_template.py

```diff
@@ -349,6 +354,46 @@ def _prepare_server_values(self, **kwargs):
 
         return vals_list
 
+    def _validate_required_variables(self, configuration_variables):
+        """
+        Validate that all required variables are present and not empty.
+
+        Args:
+            configuration_variables (dict): A dictionary of variable references
+                                             and their values.
+
+        Raises:
+            ValidationError: If any required variables are missing or empty.
+        """
+        missing_variables = []
+        empty_variables = []
+
+        for variable in self.variable_value_ids.filtered("required"):
+            variable_ref = variable.variable_reference
+            if variable_ref not in configuration_variables:
+                missing_variables.append(variable_ref)
+            elif not configuration_variables[variable_ref]:
+                empty_variables.append(variable_ref)
+
+        # Construct error message if issues are found
+        if missing_variables or empty_variables:
+            self.variable_value_ids = self.variable_value_ids
+            error_message = _(
+                "Please resolve the following issues with configuration variables:\n"
+            )
+            if missing_variables:
+                error_message += _(
+                    f"  - Missing variables: {', '.join(missing_variables)}\n"
```

Comment:
This construction may create issues with translations. Please use the recommended Odoo string translation approach. Check the DevMemo for details.